core/policy/policy.go

```diff
@@ -144,7 +148,7 @@ func (p *policyChecker) CheckPolicyNoChannel(policyName string, signedProp *pb.S
 // passed policy on the passed channel.
 func (p *policyChecker) CheckPolicyBySignedData(channelID, policyName string, sd []*protoutil.SignedData) error {
 	if channelID == "" {
-		return errors.New("Invalid channel ID name during check policy on signed data. Name must be different from nil.")
+		return p.CheckPolicyNoChannelBySignedData(policyName, sd)
```

Comment:
This function ends up being called in a ton of places, and this PR changes its failure model (prior to it, we would fail if the supplied channel ID was empty, now we don't). Have you verified that this change does not affect the verifier/committer path? Adding a new function to the 
```
type ACLProvider interface {
	//CheckACL checks the ACL for the resource for the channel using the
	//idinfo. idinfo is an object such as SignedProposal from which an
	//id can be extracted for testing against a policy
	CheckACL(resName string, channelID string, idinfo interface{}) error
}
```
interface (like a `CheckACLNoChannel` function) seems more desirable than changing the contract of the existing function, I'd go with that.

Reply:
When checking existing callers before making the change. I didn't find any code calling CheckACL with []*protoutil.SignedData. The existing code calls CheckACL with either SignedPropoal or Envelope (no []*protoutil.SignedData). However, you have a good point. It is better to add a CheckACLNoChannel and make the explicit call instead of overloading/changing existing behavior.

Reply:
Even the existing behavior of the [policy checker](https://github.com/hyperledger/fabric/blob/545754c4bab3311ca56f6098f08dd43cec7c0cf5/core/policy/policy.go#L38) is inconsistent. Two similar functions [CheckPolicy](https://github.com/hyperledger/fabric/blob/545754c4bab3311ca56f6098f08dd43cec7c0cf5/core/policy/policy.go#L51) and [CheckPolicyBySignedData](https://github.com/hyperledger/fabric/blob/545754c4bab3311ca56f6098f08dd43cec7c0cf5/core/policy/policy.go#L145) behave differently, if no channelID is specified. 

@ale-linux  - I agree that adding the exposing the function `CheckACLNoChannel` at the `ACLProvider` level makes it clearer but I would add a note (I assume that you meant this implicitly as well) that we should modify the existing callers go via this function instead of first function above. Otherwise, it will simply add to the confusion looking at the code for - which would look like
- if you work with `SignedProposal` and have no channel call the function `CheckACL` with channelID as empty string
- if you work with `SignedData`, use the function `CheckACLNoChannel`.
This modification can be done in a separate PR and can open a Jira for this.

Reply:
Yes @manish-sethi, I share your view and the approach you propose

Reply:
@manish-sethi agree, I have thought the same and will open a Jira for it.

Reply:
missed rolling back this change?